tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -77,167 +56,66 @@ setup() {
-    kata_pod_with_nydus_config="$(new_pod_config "$large_image" "kata-${KATA_HYPERVISOR}")"
-    set_node "$kata_pod_with_nydus_config" "$node"
-    set_container_command "$kata_pod_with_nydus_config" "0" "sleep" "30"
+@test "Test we can pull a large image inside the guest using trusted ephemeral storage" {
+    
+    # The image pulled in the guest will be downloaded and unpacked in the `/run/kata-containers/image` directory. 
+    # However, by default, systemd allocates 10% of the available physical RAM to the `/run` directory using a `tmpfs` filesystem. 
+    # It means that if we run a kata container with the default configuration (where the default memory assigned for a VM is 2048 MiB), 
+    # `/run` would be allocated around 200 MiB. Consequently, we can only pull images up to 200 MiB in the guest. 
+    # And the unpacked size of image "ghcr.io/confidential-containers/test-container:big-size" is 965MB. 
+    # It will fail to run the pod with pulling the image in the memory in the guest by default. 
 
-    # Set annotation to pull large image in guest
-    set_metadata_annotation "$kata_pod_with_nydus_config" \
-        "io.containerd.cri.runtime-handler" \
-        "kata-${KATA_HYPERVISOR}"
+    pod_config="$(new_pod_config "$large_image" "kata-${KATA_HYPERVISOR}")"
+    set_node "$pod_config" "$node"
+    set_container_command "$pod_config" "0" "sleep" "30"
 
     # For debug sake
-    echo "Pod $kata_pod_with_nydus_config file:"
-    cat $kata_pod_with_nydus_config
+    echo "Pod $pod_config file:"
+    cat $pod_config
 
     # The pod should be failed because the default timeout of CreateContainerRequest is 60s
-    assert_pod_fail "$kata_pod_with_nydus_config"
+    assert_pod_fail "$pod_config"
     assert_logs_contain "$node" kata "$node_start_time" \
 		'context deadline exceeded'
 
-    kubectl delete -f $kata_pod_with_nydus_config
-
-    # Set CreateContainerRequest timeout in the annotation to pull large image in guest
-    create_container_timeout=300
-    set_metadata_annotation "$kata_pod_with_nydus_config" \
-        "io.katacontainers.config.runtime.create_container_timeout" \
-        "${create_container_timeout}"
-
-    # For debug sake
-    echo "Pod $kata_pod_with_nydus_config file:"
-    cat $kata_pod_with_nydus_config
-
-    add_allow_all_policy_to_yaml "$kata_pod_with_nydus_config"
-    k8s_create_pod "$kata_pod_with_nydus_config"
-}
-
-@test "Test we can pull an unencrypted image inside the guest twice in a row and then outside the guest successfully" {
-    # 1. Create one kata pod with the $unencrypted_image_1 image and nydus annotation twice
-    kata_pod_with_nydus_config="$(new_pod_config "$unencrypted_image_1" "kata-${KATA_HYPERVISOR}")"
-    set_node "$kata_pod_with_nydus_config" "$node"
-    set_container_command "$kata_pod_with_nydus_config" "0" "sleep" "30"
+    kubectl delete -f $pod_config
```

Comment:
Do you think it would be better to split the positive and negative cases into two tests? We have done that in some of our other files.

Reply:
Sure.

Reply:
Done.